src/Test/Utilities/Portable/Compilation/IRuntimeEnvironment.cs

```diff
@@ -27,20 +27,13 @@ internal static IRuntimeEnvironment Create(IEnumerable<ModuleData> additionalDep
 
         private static IRuntimeEnvironmentFactory GetFactoryImplementation()
         {
-            string assemblyName;
-            string typeName;
-            if (CoreClrShim.IsRunningOnCoreClr)
-            {
-                assemblyName = "Roslyn.Test.Utilities.CoreClr";
-                typeName = "Microsoft.CodeAnalysis.Test.Utilities.CodeRuntime.CoreCLRRuntimeEnvironmentFactory";
-            }
-            else
-            {
-                assemblyName = "Roslyn.Test.Utilities.Desktop";
-                typeName = "Microsoft.CodeAnalysis.Test.Utilities.CodeRuntime.DesktopRuntimeEnvironmentFactory";
-            }
-
```

Comment:
Our current build strategy with runtime depednencies doesn't work with how deployment works today. In order for the runtime check to work the TestUtil.CoreClr or TestUtil.Desktop must be present in the output directory. 

This works today because we use a desktop / coreclr deployment project that references these projects and acts as a deployment mechanism. The model we need to move to is where build / deployment is self sufficient for test projects. Hence our portable test projects must reference that project at build or deployment time. 

Referencing at build time would be a possibility if `<ProjectReference>` supported the `Conditional` attribute. Then we could essentially add the following to every portable test project

``` xml
<ProjectReference Include="TestUtilities.CoreClr.csproj" Condition="'$(TargetFramework)' == 'netcoreapp2.0'" />
<ProjectReference Include="TestUtilities.Desktop.csproj" Condition="'$(TargetFramework)' == 'net46'" />
```

I chatted with the SDK team about this and it's not supported now. It's also unlikely to be supported anytime soon. 

After a lot of discussion the most approachable plan is what this PR represents. Multi-target the core utility project and the final test projects which are logically portable. This means build + deployment will pick the right code for each target framework.

In the future once we support deployment of netstandard projects we can remove the mulit-targeting on our test projects and simply target netstandard1.3. That's at least a month or two away though. 

Yes we do run more compilations but this is the only path supported by our tooling. The extra compilations don't really seem to factor in too much here. They did in my original attempt which had a lot more multi-targeted projects. But this amount seems pretty negliable.